### ISSUE TITLE
[Installation with pip] No section: 'params' error

### ISSUE THREAD

**Issue by Rmulet** [vanheeringen-lab/gimmemotifs#59]
Hi Simon,

I have been trying to use gimmemotifs lately, with little success. With `conda create`, my installation is stuck forever on 'Solving package specifications' unless I specify the version of gimmemotifs I want, something I took from a post of yours in the bioconda GitHub. Leaving that aside, what I want to report here is an error whenever I try to use gimmemotifs following its installation via `pip`. Namely, I get the following:

> Traceback (most recent call last):
>   File "/usr/lib/python3.5/configparser.py", line 843, in items
>     d.update(self._sections[section])
> KeyError: 'params'
> 
> During handling of the above exception, another exception occurred:
> 
> Traceback (most recent call last):
>   File "/data/roger/.local/bin/gimme", line 11, in <module>
>     from  gimmemotifs import commands
>   File "/data/roger/.local/lib/python3.5/site-packages/gimmemotifs/commands/__init__.py", line 14, in <module>
>     m =  __import__("{0}.{1}".format(__name__, cmdname), globals(), locals(), [cmdname], level)
>   File "/data/roger/.local/lib/python3.5/site-packages/gimmemotifs/commands/location.py", line 14, in <module>
>     from gimmemotifs.mp import pool
>   File "/data/roger/.local/lib/python3.5/site-packages/gimmemotifs/mp.py", line 5, in <module>
>     n_cpus = int(MotifConfig().get_default_params()["ncpus"])
>   File "/data/roger/.local/lib/python3.5/site-packages/gimmemotifs/config.py", line 66, in get_default_params
>     d = dict(self.config.items("params"))
>   File "/usr/lib/python3.5/configparser.py", line 846, in items
>     raise NoSectionError(section)
> configparser.NoSectionError: No section: 'params'

I suspect it may have something to do with the configuration file, but strangely enough, I did not encounter this problem when I installed the program as a part of a conda environment. Am I doing anything wrong or is it really an error? Thanks!

**Comment by simonvh** [comment]
I suspect it's not something you're doing wrong, but an issue with pip that I don't yet understand. Have you tried this command:

```
pip install git+https://github.com/simonvh/gimmemotifs@0.11.1
```

Instead of (what used to be in my docs):

```
pip install https://github.com/simonvh/gimmemotifs/releases/download/0.11.1/gimmemotifs-0.11.1.tar.gz
```
The last one gives this error, the first `pip` command should work fine. Let me know if it doesn't!

(And the conda-getting-stuck-problem is also frustrating to me as I don't know how to solve it.)

**Comment by Rmulet** [comment]
Hi, I tried the suggested command, but I get the same error. There is apparently no difference between Python3 and Python2.

**Comment by simonvh** [comment]
If I interpret the code correctly you have installed via pip outside of a conda environment, right? Can you give me the exact steps/commands you used for installation? This might help me to solve it. What is the OS?

**Comment by Rmulet** [comment]
Yes, that is indeed the case. I tried to install `gimmemotifs` in different ways, all of them outside a Conda environment, but it made no difference. I simply used pip, as shown in the examples before retrieved from my bash history:

> pip3 install --user git+https://github.com/simonvh/gimmemotifs@0.11.1
> sudo pip3 install git+https://github.com/simonvh/gimmemotifs@0.11.1

On the other hand, I have managed to successfully install this same version (0.11) on a Conda environment following the instructions that you provide in this repository. 

As for the OS, I am using Ubuntu 16.04 LTS.

**Comment by simonvh** [comment]
Better late than never, this issue should be fixed now. The conda problem (where it got stuck during dependency calculation) seems to have disappeared, with no apparent involvement from my side. This means that installing gimmemotifs with conda should be a one-step procedure again.

The pip issue was much more complicated and had to do with both how setuptools and pip interact as well as with the way I implemented the config file. This has been completely overhauled, and starting from the next release (or the git develop branch) it should be possible to install with pip resulting in a working config. In any case, thanks for reporting this issue.

**Comment by simonvh** [comment]
Should be fixed in 0.13.0.
